components/primitives/Form/Input.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { cn } from "@/lib/utils/cn";
+
+interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {}
+
+const Input = React.forwardRef<HTMLInputElement, InputProps>(
+  ({ id, type, className, ...rest }, ref) => {
+    const [showPassword, setShowPassword] = React.useState(false);
```

Comment:
We should also do away with `state` management within `primitives`, that's the aim for grouping these cateogories of component differently though. 

They shouldn't manage any local state at all.... You can fulfil this functionality with the `ref` instead.

Reply:
ref still uses hooks Is there a reason we don't want to use state I see no difference even if we are achieving it with ref just extra stress for little to no advantage what do you think I can change it tho @289Volts

Reply:
> ..do away with `state` management within `primitives`...

We should do away with managing local states within these category of components at every opportunity that we get, that is the pattern that we are tryna adopt, haha 😃 

Kindly make the changes, it even mean less code written you see!? 🤲

Reply:
We would we be doing away with hooks but now we are changing the real DOM which according to React was meant to be an escape hatch but I will do it if that the way we are going